presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayEqualOperator.java

```diff
@@ -56,10 +55,6 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
 
     public static boolean equals(Type type, Slice left, Slice right)
     {
-        BlockBuilder leftBlockBuilder = type.createBlockBuilder(new BlockBuilderStatus());
-        BlockBuilder rightBlockBuilder = type.createBlockBuilder(new BlockBuilderStatus());
-        leftBlockBuilder.writeBytes(left, 0, left.length());
-        rightBlockBuilder.writeBytes(right, 0, right.length());
-        return type.equalTo(leftBlockBuilder.closeEntry().build(), 0, rightBlockBuilder.closeEntry().build(), 0);
+        return type.equalTo(createBlock(type, left), 0, createBlock(type, right), 0);
```

Comment:
In this, `type` is ArrayType so `ArrayType.equalTo` could check each elements by passing it as a whole with `createBlock`. If `ArrayBuilder.readBlock` is used, it might have to duplicate what `ArrayType.equalTo` does.